### PR TITLE
Do not check uniqueness of parameter names twice

### PR DESCRIPTION
**Pull request by alrz**
No observable changes.

### CODE REVIEW

**Comment by alrz** [comment]
@sharwell May I have this under **Area-Compilers** for the team to take a look? thanks.

**Comment by Pilchie** [comment]
Tagging @dotnet/roslyn-compiler for reviews.

File: src/Compilers/CSharp/Portable/Binder/Binder_NameConflicts.cs [hunk 1]
```diff
@@ -76,14 +75,13 @@ private static Location GetLocation(Symbol symbol)
                         diagnostics.Add(ErrorCode.ERR_LocalSameNameAsTypeParam, GetLocation(p), name);
                     }
 
-                    if (pNames.Contains(name))
+                    if (!pNames.Add(name))
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Binder/Binder_NameConflicts.cs]: We should really write an analyzer to catch patterns like this.

**Comment by alrz** [comment]
I've found some other instances of this, will commit to this PR.

**Comment by jcouv** [comment]
@dotnet-bot retest ubuntu_16_debug_prtest  please

**Comment by jcouv** [comment]
retest windows_debug_vs-integration_prtest  please

File: src/Compilers/CSharp/Portable/Binder/Binder_NameConflicts.cs (hunk 1 above)
Comment (VSadov) [inline src/Compilers/CSharp/Portable/Binder/Binder_NameConflicts.cs]: hashset.Add with unused result is often a suspect.

**Pull request merged**